packages/e2e-tests/helpers.ts

```diff
@@ -790,3 +800,110 @@ export async function clickSourceTreeNode(screen: Screen, node: string) {
   debugPrint(`Selecting source tree node: ${chalk.bold(node)}`);
   await screen.locator(`div[role="tree"] div:has-text("${node}")`).nth(1).click();
 }
+
+export function getMarkupNode(screen: Screen, text: string) {
+  const escapedText = text.replace(/\\/g, "\\\\").replace(/"/g, '\\"');
+  return screen.locator(`#markup-box .editor:has-text("${escapedText}")`);
+}
+
+export async function selectMarkupNode(screen: Screen, text: string) {
+  const node = getMarkupNode(screen, text);
+  await node.waitFor();
+  await node.click();
+}
+
+export async function toggleMarkupNode(locator: Locator) {
+  await locator.evaluate(node => {
+    const expander = node.parentElement?.querySelector(".expander");
+    if (expander) {
+      (expander as HTMLElement).click();
+    }
+  });
+}
+
+export function waitForSelectedMarkupNode(screen: Screen, text: string) {
+  const escapedText = text.replace(/\\/g, "\\\\").replace(/"/g, '\\"');
+  return screen.locator(`#markup-box .selected > .editor:has-text("${escapedText}")`).waitFor();
+}
+
+export async function searchMarkup(screen: Screen, text: string) {
+  const searchBox = screen.getByPlaceholderText("Search HTML");
+  await searchBox.fill(text);
+  await searchBox.press("Enter");
+}
+
+export async function selectNextMarkupSearchResult(screen: Screen) {
+  const searchBox = screen.getByPlaceholderText("Search HTML");
+  await searchBox.press("Enter");
+}
+
+export async function checkComputedStyle(
+  screen: Screen,
+  style: string,
+  value: string,
+  expectedSelectors: any = undefined
+) {
+  await selectComputedProperties(screen);
+  await waitFor(async () => {
+    const result = await screen.evaluate(
+      ({ style, value, matchedSelectors }) => {
+        function getMatchedSelectors(property: string) {
+          const propertyNodes = document.querySelectorAll<HTMLElement>(".computed-property-view");
+
+          const propertyNode = [...propertyNodes].find(
+            node =>
+              node!.querySelector(".computed-property-name")!.childNodes[0].textContent === property
+          );
+
+          if (!propertyNode) {
+            return [];
+          }
+
+          const expander: HTMLElement | null = propertyNode.querySelector(".computed-expander");
+          if (!expander!.matches(".open")) {
+            expander!.click();
+          }
+
+          const selectorNodes = (
+            propertyNode.nextSibling as HTMLElement
+          ).querySelectorAll<HTMLElement>(".rule-text");
+          return [...selectorNodes].map(selectorNode => {
+            const selector = (selectorNode.children[0] as HTMLElement).innerText;
+            const value = (selectorNode.children[1] as HTMLElement).innerText;
+            const label = (selectorNode.previousSibling as HTMLElement).innerText;
+            const previousChild = (selectorNode.previousSibling as HTMLElement).children[0];
```

Comment:
Some of this method, like this section with [0] and [1] children, looks kind of brittle. I wonder if there's a way to rewrite this that isn't so dependent on the _order_ of children (which could easily change)

Reply:
I'm sure this could be improved. I'm also not 100% certain that having all this logic inside an `evalute()` is optimal. But it's good enough for now (and we want to migrate the tests ASAP).